docs/src/userguide/language_basics.rst

```diff
@@ -381,6 +381,48 @@ Sometimes Cython will complain unnecessarily, and sometimes it will fail to
 detect a problem that exists. Ultimately, you need to understand the issue and
 be careful what you do.
 
+
+Type Casting
+------------
+
+As C uses the ``"("`` and ``")"`` for casting types, Cython uses ``"<"`` and ``">"``
+to signify a casting. For example::
+
+    cdef char *p
+    cdef float *q
+    p = <char*>q
+
+If one of the types is a Python object for ``<type>x``, Cython will try to do a coercion.
+
+ .. note:: Cython will not stop a casting where there is no conversion, but it will emit a warning.
+
+To get the address of some Python object, use a cast to a pointer type
+like ``<void*>`` or ``<PyObject*>``.
+You can also get the python object from a pointer to this object by doing ``<object>``.
+Here is an example::
+
+    from __future__ import print_function
+    from cpython.ref cimport PyObject
+
+    python_string = "foo"
+
+    cdef void* ptr = <void*>python_string
+    cdef long adress_in_c = <long>ptr
```

Comment:
Bad example since this is not portable. Pointers do not necessarily fit into the size of a `long`, e.g. on Win64.
You can cimport `uintptr_t` from `libc.stdint`, though, and use that.